.github/workflows/package-workflow.yml

```diff
@@ -51,5 +81,79 @@ jobs:
         run: yarn workspace ${{ env.PROJECT_NAME }} package
       - uses: actions/upload-artifact@v2
         with:
-          name: ${{ env.PROJECT_NAME }}
-          path: packages/${{ env.PROJECT_NAME }}/vsce/*.vsix
+          name: ${{ env.PROJECT_NAME }}-${{ env.PACKAGE_VERSION }}
+          path: ${{ env.VSIX_PATH }}
+
+      - name: Create GitHub release
+        if: ${{ github.event_name == 'push' }} && steps.audit.outputs.exit_code < 7
```

Comment:
I propose to separate two workflows into two different files. 

The first one will be for all branches: building, testing, packaging and etc.
The second one will be for releasing (only on master branch).

Now, we have everything is one file and we have to create a complicated `if` statements for each step...

What do you think?

Reply:
And the releasing workflow will be super simple, without any matrix or smt like this. We can just use `ubuntu`.

Also, it would be great to run that separate workflow only in case of successful execution of the usual workflow for all the branches.

Reply:
Ok, I've tried to split it, let's see if it runs this time.